icu4c/source/test/intltest/loctest.cpp

```diff
@@ -4101,6 +4101,34 @@ LocaleTest::TestCreateUnicodeKeywordSet(void) {
                result.find("ca") != result.end());
     assertTrue("set::find(\"co\")",
                result.find("co") != result.end());
+
+    static const char tag[] = "en-US-u-ca-gregory-x-foo";
+    Locale l2 = Locale::forLanguageTag(tag, status);
+    result.clear();
+    l2.getKeywords<std::string>(
+            std::insert_iterator<decltype(result)>(result, result.begin()),
+            status);
+    status.errIfFailureAndReset("getKeywords \"%s\"", l2.getName());
+    assertTrue("getKeywords set::find(\"calendar\")",
+               result.find("calendar") != result.end());
+    assertTrue("getKeywords set::find(\"ca\")",
+               result.find("ca") == result.end());
+    assertTrue("getKeywords set::find(\"x\")",
+               result.find("x") != result.end());
+    assertTrue("getKeywords set::find(\"co\")",
+               result.find("co") == result.end());
+
+    result.clear();
+    l2.getUnicodeKeywords<std::string>(
+            std::insert_iterator<decltype(result)>(result, result.begin()),
+            status);
+    status.errIfFailureAndReset("getUnicodeKeywords \"%s\"", l2.getName());
+    assertTrue("getUnicodeKeywords set::find(\"ca\")",
+               result.find("ca") != result.end());
+    assertTrue("getUnicodeKeywords set::find(\"x\")",
+               result.find("x") == result.end());
+    assertTrue("getUnicodeKeywords set::find(\"co\")",
```

Comment:
See previous comment, `"foo"` instead of `"co"`, but just as the existing test code for `getUnicodeKeywords()` skips this entirely because it has a test for `size()` instead, I think that's what you'd better do here too.

Reply:
fixed